src/codeu/chat/client/simplegui/UserPanel.java

```diff
@@ -164,8 +164,16 @@ public void actionPerformed(ActionEvent e) {
             UserPanel.this, "Enter user name:", "Add User", JOptionPane.PLAIN_MESSAGE,
             null, null, "");
         if (s != null && s.length() > 0) {
-          clientContext.user.addUser(s);
-          UserPanel.this.getAllUsers(listModel);
+          //if alphanumeric
+          if(s.matches("[a-zA-Z0-9]+")) {
```

Comment:
Patterns stored across classes that need to be kept in sync can go out of sync easily. Maybe a central common constant class?